indexer_util/indexer_util/indexer_util.py

```diff
@@ -0,0 +1,90 @@
+"""Utilities for Data Explorer indexers"""
+
+import csv
+import jsmin
+import json
+import logging
+import os
+import time
+
+from elasticsearch import Elasticsearch
+from elasticsearch.exceptions import ConnectionError
+
+# Log to stderr.
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s %(filename)10s:%(lineno)s %(levelname)s %(message)s',
+    datefmt='%Y%m%d%H:%M:%S')
+logger = logging.getLogger('indexer.util')
+
+ES_TIMEOUT_SEC = 20
+
+
+# Copied from https://stackoverflow.com/a/45392259
+def environ_or_required(key):
+    if os.environ.get(key):
+        return {'default': os.environ.get(key)}
+    else:
+        return {'required': True}
+
+
+def open_and_return_json(file_path):
+    """Opens and returns JSON contents.
+
+  Args:
+    file_path: Relative path of JSON file.
+
+  Returns:
+    Parsed JSON.
+  """
+    with open(file_path, 'r') as f:
+        # Remove comments using jsmin, as recommended by JSON creator
+        # (https://plus.google.com/+DouglasCrockfordEsq/posts/RK8qyGVaGSr).
+        jsonDict = json.loads(jsmin.jsmin(f.read()))
+        return jsonDict
+
+
+# Keep in sync with convert_to_index_name() in data-explorer repo.
+def convert_to_index_name(s):
+    """Converts a string to an Elasticsearch index name."""
+    # For Elasticsearch index name restrictions, see
+    # https://github.com/DataBiosphere/data-explorer-indexers/issues/5#issue-308168951
+    prohibited_chars = [' ', '"', '*', '\\', '<', '|', ',', '>', '/', '?']
+    for char in prohibited_chars:
+        s = s.replace(char, '_')
+    s = s.lower()
+    # Remove leading underscore.
+    if s.find('_', 0, 1) == 0:
+        s = s.lstrip('_')
+    print('Index name: %s' % s)
+    return s
+
+
+def init_elasticsearch(elasticsearch_url, index_name):
```

Comment:
I think `init_elasticsearch` makes sense. This method performs Elasticsearch initialization: Start it, wait for it to be healthy, and create index. Added method pydoc. I don't see why 1 and 2 should be moved to a separate method.

Reply:
If I'm understanding correctly, this method doesn't actually start Elasticsearch, it just instantiates an Elasticsearch client that references an already running Elasticsearch server. This was an important point for me to understand reading this code

Reply:
Oops, you're right. Updated comment. I still prefer `init_elasticsearch` over `create_elasticsearch_index`. Creating the index is one thing it does; the other is waiting for ES to come up. Or I'm open to other suggestions.

Reply:
That's why I suggested that `create_es_index` call a `wait_for_es` helper method, since the purpose of this function is to create the index, and waiting for server health is more of a how.

Reply:
Done